File: apps/gustav/src/lib/env.ts
// Build-time configuration, read from EXPO_PUBLIC_* env vars (see
// .env.example) — never hardcoded secrets. The anon key is meant to ship in
// the client; the service-role key never does and has no place here.
export const env = {
  supabaseUrl: process.env.EXPO_PUBLIC_SUPABASE_URL ?? 'http://127.0.0.1:54321',
  supabaseAnonKey: process.env.EXPO_PUBLIC_SUPABASE_ANON_KEY ?? '',
} as const;

File: apps/gustav/lib/app/env.dart
/// Build-time configuration — never hardcoded secrets, always passed in via
/// `--dart-define` (see `infra/supabase/.env.example` for where the values
/// come from). The anon key is meant to ship in the client (Project
/// Settings → API); the service-role key never does and has no place here.
class Env {
  const Env._();

  static const supabaseUrl = String.fromEnvironment(
    'SUPABASE_URL',
    defaultValue: 'http://127.0.0.1:54321',
  );

  static const supabaseAnonKey = String.fromEnvironment(
    'SUPABASE_ANON_KEY',
    defaultValue: '',
  );
}
